pkg/spec/spec.go

```diff
@@ -74,18 +74,23 @@ func CreateConfigToOCISpec(config *CreateConfig) (*spec.Spec, error) { //nolint
 		g.AddAnnotation(key, val)
 	}
 	g.SetRootReadonly(config.ReadOnlyRootfs)
-	if config.Hostname == "" {
-		if config.NetMode.IsHost() {
-			config.Hostname, err = os.Hostname()
-			if err != nil {
-				return nil, errors.Wrap(err, "unable to retrieve hostname")
-			}
+
+	hostname := config.Hostname
+	if hostname == "" && (config.NetMode.IsHost() || config.UtsMode.IsHost()) {
```

Comment:
Possible suggestion for later.  If we end up needing this double "*.IsHost()" check in more places than this, or we add more *.IsHost() functions in the future, it might be nice to have a little "isHost()" utils function that would return a true boolean if any of them are true.  Nothing to worry about now, just something for consideration.